vidyut-lipi/src/lib.rs

```diff
@@ -1,118 +1,551 @@
-        "ṝ" => "F",
-        "ḷ" => "x",
-        "ḹ" => "X",
-        "ai" => "E",
-        "au" => "O",
-        "ṃ" => "M",
-        "ḥ" => "H",
-        "ṅ" => "N",
-        "kh" => "K",
-        "gh" => "G",
-        "ch" => "C",
-        "jh" => "J",
-        "ñ" => "Y",
-        "ṭ" => "w",
-        "ṭh" => "W",
-        "ḍ" => "q",
-        "ḍh" => "Q",
-        "th" => "T",
-        "dh" => "D",
-        "ph" => "P",
-        "bh" => "B",
-        "ṇ" => "R",
-        "ś" => "S",
-        "ṣ" => "z",
-        "ḻ" => "L",
-        &_ => return None,
-    };
-    Some(val)
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Scheme {
+    encodings: Vec<Encoding>,
+    implicit: String,
 }
 
-/// Hackily transliterate from IAST to SLP1.
-fn iast_to_slp1(input: &str) -> String {
-    let chars: Vec<char> = input.chars().collect();
-    let mut ret = String::new();
-    let mut i = 0;
-    while i < chars.len() {
-        let mut next: Option<&str> = None;
-        let mut offset = 0;
-
-        // Search for matches against our mapping. The longest IAST glyph has two characters,
-        // so search up to length 2. Start with 2 first so that we match greedily.
-        for j in [2, 1] {
-            let limit = cmp::min(i + j, chars.len());
-            let cur = String::from_iter(&chars[i..limit]);
-            offset = limit - i;
-
-            next = map_char(cur.as_str());
-            if let Some(_s) = next {
-                break;
-            }
-        }
+// Node represents the input mnemonic.
+// mnem: string associated with the node. ex: a, k or kh
+// tok_val: token associated with the mnemonic in a scheme
+// suffix: indicates whether the current mnem is a prefix to another mnem
+// tok_type: is it consonant or vowel. Depending on the scheme we may need
+//           pop or push a character.
+// implicit: this field may not be required - since info is already in
+//           the scheme.
+#[allow(dead_code)]
+#[derive(Clone, Debug, Default)]
+pub struct Node {
+    mnem: String, // Input mnemonic
+    #[allow(dead_code)]
+    assoc: String, // If a vowel - then the associated sign.
+    // tok_val was initially planned as a u8. However 
+    // using a 
+    tok_val: String,  // Token value associated with the string.
+    // Is there a possible suffix? If so need to check
+    // the suffix as well since longest match is required.
+    suffix: bool,
+    // Is this token a vowel or consonant
+    tok_type: u8,
+    // Is there an implicit virama or a?
+    implicit: u8,
+}
+
+impl Node {
+    #[allow(dead_code)]
+    fn default() -> Self {
+	Node {
+	    mnem: "".to_string(),
+	    assoc: "".to_string(), // For vowels - associated sign. 
+	    tok_val: "TOKEN_TRANSPARENT".to_string(), // Pass input to output transparently. 
+	    suffix: false,
+	    tok_type: TOK_UNKNOWN_TYPE,
+	    implicit: IMPLICIT_UNKNOWN,
+	}
+    }
+    fn init(m:String, a:String, tv:String,tt:u8,suf:bool,imp:u8) -> Self {
+	Node {
+	    mnem: m,
```

Comment:
Use the same names in your parameters. Then you can do:

```
Node { mnem, assoc, tok_val, tok_type, suffix, implicit }
```

Reply:
ok.